src/emqx_acl_cache.erl

```diff
@@ -67,17 +68,22 @@ list_acl_cache() ->
 
 %% We'll cleanup the cache before replacing an expired acl.
 -spec(get_acl_cache(emqx_types:pubsub(), emqx_topic:topic()) -> (acl_result() | not_found)).
-get_acl_cache(PubSub, Topic) ->
-    case erlang:get(cache_k(PubSub, Topic)) of
-        undefined -> not_found;
-        {AclResult, CachedAt} ->
-            if_expired(CachedAt,
-                fun(false) ->
-                      AclResult;
-                   (true) ->
-                      cleanup_acl_cache(),
-                      not_found
-                end)
+get_acl_cache(PubSub, Topic)  ->
```

Comment:
We try to avoid nested `case` expressions to keep sane indentation levels.

If we change this function to accept a new args:
```suggestion
get_acl_cache(PubSub, Topic, _IsDiscarded = true)  ->
    _ = empty_ack_acche(),
    not_found;
```
then and keep the `false` clause untouched.

Which implies we need to delegate the /2 call to /3 like this:

```
get_acl_cache(PubSub, Topic) -> get_acl_cache(PubSub, Topic, is_discarded()).
```